File: app/src/main/java/com/androidexamples/app/adapter/IntroAdapter.java
package com.androidexamples.app.adapter;

import android.content.Context;
import android.support.v4.view.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import java.util.List;

/**
 * Created by marco on 19/01/2017.
 */

public class IntroAdapter extends PagerAdapter {

    private Context context;
    private List<Integer> listLayouts;

    public IntroAdapter(Context context, List<Integer> listLayouts) {
        this.context = context;
        this.listLayouts = listLayouts;
    }

    @Override
    public Object instantiateItem(ViewGroup container, int position) {
        View view = LayoutInflater.from(context).inflate(listLayouts.get(position), container, false);
        container.addView(view);
        return view;
    }

    @Override
    public int getCount() {
        return listLayouts.size();
    }

    @Override
    public boolean isViewFromObject(View view, Object object) {
        return view == object;
    }

    @Override
    public void destroyItem(ViewGroup container, int position, Object object) {
        View view = (View)object;
        container.removeView(view);
    }
}

File: app/src/main/java/com/androidexamples/app/adapter/IntroAdapter.kt
package com.androidexamples.app.adapter

import android.content.Context
import android.support.v4.view.PagerAdapter
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup

/**
 * Created by marco on 19/01/2017.
 */

class IntroAdapter(private val context: Context, private val listLayouts: List<Int>) : PagerAdapter() {

    override fun instantiateItem(container: ViewGroup, position: Int): Any {
        val view = LayoutInflater.from(context).inflate(listLayouts[position], container, false)
        container.addView(view)
        return view
    }

    override fun getCount(): Int {
        return listLayouts.size
    }

    override fun isViewFromObject(view: View, `object`: Any): Boolean {
        return view === `object`
    }

    override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
        val view = `object` as View
        container.removeView(view)
    }
}
